src/Session.js

```diff
@@ -38,14 +38,24 @@ const Session = class Session {
         });
     }
 
-    markAccessed(modelName) {
-        this.getDataForModel(modelName).accessed = true;
+    markAccessed(modelName, modelIds) {
+        const data = this.getDataForModel(modelName);
+        if (!data.accessed) {
+            data.accessed = [];
+        }
+        data.accessed.push(...modelIds);
```

Comment:
My recommendation would be to start this off as a hash, not an array. Could be just a hash of IDs => true/false.

Making an array is probably better for memory, but to @plandem's point, if we have 10k+ objects, imagine that the @connect function is doing a state compare on those 10000 objects, it has to literally loop through each component, and run a search for the instance IDs in the array of 10000 objects and *then* has to go through the other 10000 components on the page and do the same thing. This will be slow.

For this reason, I say, go with the object lookup approach first. The users of your website will thank you... :)